File: backend/src/main/java/it/unicam/dmr/doit/utenti/ruoli/Proponente.java
package it.unicam.dmr.doit.utenti.ruoli;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;

import it.unicam.dmr.doit.progetto.Progetto;

/**
 * Questa classe estende {@code Ruolo} e rappresenta il ruolo
 * {@code TipologiaRuolo.ROLE_PROPONENTE}.
 * 
 * @author Damiano Cacchiarelli
 * @author Matteo Romagnoli
 * @author Roberto Cesetti
 */
@Entity
public class Proponente extends Ruolo {
		
	@OneToMany(mappedBy = "proponente", fetch = FetchType.LAZY,
            cascade = CascadeType.ALL)
	private Set<Progetto> proposte = new HashSet<>();
	
	public Proponente() {
		setRuolo(TipologiaRuolo.ROLE_PROPONENTE);
	}
	
	@Override
	public Set<Progetto> getProgettiPersonali() {
		return proposte;
	}
}

File: frontend/src/app/modello/iscritto/ruolo/proponente.ts
import { Progetto } from "../../progetto/progetto";
import { Ruolo } from "./ruolo";
import { TipologiaRuolo } from "./tipologia-ruolo.enum";

export class Proponente extends Ruolo {


    constructor(ruolo: TipologiaRuolo, id: number, progettiPersonali: Progetto[]) {

        super(ruolo, id, progettiPersonali);
    }

}